components/kserve/kserve_config_handler.go

```diff
@@ -25,18 +25,18 @@ const (
 func (k *Kserve) setupKserveConfig(ctx context.Context, cli client.Client, logger logr.Logger, dscispec *dsciv1.DSCInitializationSpec) error {
 	// as long as Kserve.Serving is not 'Removed', we will setup the dependencies
 
+	defaultDeploymentMode := k.DefaultDeploymentMode
```

Comment:
In reality, this is changing the outcome and doesn't look right.
**Previously**
* When Knative is available:
  * If the user does not specify a default deployment mode:
    * We configure **Serverless** as the default mode.
    * We **enable** creation of the ingress.
  * If the user specifies a default mode:
    * We **respect** the user specified default deployment mode.
    * We **disable** creation of the ingress if the user specified Raw. Otherwise, we **enable** it.
* When Knative is NOT available:
  * If the user does not specify a default deployment mode:
    * We configure **Raw** as the default mode.
    * We **disable** creation of the ingress.
  * If the user specifies a default mode:
    * If the user specified _Serverless_ as default mode, we stop reconciliation with an error.
    * Because of the previous, the user can only specify Raw. So, we use Raw.
    * We **disable** creation of the ingress.

**Now**
* When Knative is available:
  * If the user does not specify a default deployment mode:
    * We configure **Serverless** as the default mode.
    * We **disable** the ingress if the user didn't specify a config. Otherwise, we use the user config.
      * _The default is different from the previous state, as there is no conditional._
  * If the user specifies a default mode:
    * We **respect** the user specified default deployment mode.
    * We **disable** the ingress if the user didn't specify a config. Otherwise, we use the user config.
       * _The default is different from the previous state, as there is no conditional._
* When Knative is NOT available:
  * If the user does not specify a default deployment mode:
    * We take **Serverless** as the default mode. Then, we return an error with our own default.
      * _This is quite wrong, IMO._
  * If the user specifies a default mode:
    * If the user specified _Serverless_ as default mode, we stop reconciliation with an error.
    * Because of the previous, the user can only specify Raw. So, we use Raw.
    * We **disable** the ingress if the user didn't specify a config. Otherwise, we use the user config.
      * This one looks OK.

Generally speaking, any upgrade of a previous install will receive an updated config which, IMO, shouldn't happen. We should be backwards compatible here.